pkg/helm/handlers/handler_test.go

```diff
@@ -290,7 +330,202 @@ func TestHelmHandlers_HandleGetChart(t *testing.T) {
+			name:                "Return chart info in json format",
+			expectedResponse:    `[{"name":"test","version":1},{"name":"test","version":2}]`,
+			history:             fakeReleaseHistory,
+			httpStatusCode:      http.StatusOK,
+			expectedContentType: "application/json",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handlers := fakeHelmHandler()
+			handlers.getReleaseHistory = fakeGetReleaseHistory(tt.history, tt.error)
+
+			request := httptest.NewRequest("", "/foo", strings.NewReader("{}"))
+			response := httptest.NewRecorder()
+
+			handlers.HandleGetReleaseHistory(&auth.User{}, response, request)
+
+			if response.Code != tt.httpStatusCode {
+				t.Errorf("response code should be %v but got %v", tt.httpStatusCode, response.Code)
+			}
+			if response.Header().Get("Content-Type") != tt.expectedContentType {
+				t.Errorf("content type should be %s but got %s", tt.expectedContentType, response.Header().Get("Content-Type"))
+			}
+			if response.Body.String() != tt.expectedResponse {
+				t.Errorf("response body not matching expected is %s and received is %s", tt.expectedResponse, response.Body.String())
+			}
+		})
+	}
+}
+
+func TestHelmHandlers_HandleHelmUninstallRelease(t *testing.T) {
+	tests := []struct {
+		name                string
+		expectedResponse    string
+		uninstallResponse   *release.UninstallReleaseResponse
+		expectedContentType string
+		error
+		httpStatusCode int
+	}{
+		{
+			name:                "Error occurred",
+			error:               errors.New("Chart path is invalid"),
+			expectedResponse:    `{"error":"Failed to uninstall helm release: Chart path is invalid"}`,
+			httpStatusCode:      http.StatusBadGateway,
+			expectedContentType: "application/json",
+		},
+		{
+			name:                "Return chart info in json format",
+			expectedResponse:    `{"release":{"name":"Test"}}`,
+			uninstallResponse:   fakeUninstallResponse,
+			httpStatusCode:      http.StatusOK,
+			expectedContentType: "application/json",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handlers := fakeHelmHandler()
+			handlers.uninstallRelease = fakeUninstallRelease(tt.uninstallResponse, tt.error)
+
+			request := httptest.NewRequest("", "/foo", strings.NewReader("{}"))
+			response := httptest.NewRecorder()
+
+			handlers.HandleHelmUninstallRelease(&auth.User{}, response, request)
+
+			if response.Code != tt.httpStatusCode {
+				t.Errorf("response code should be %v but got %v", tt.httpStatusCode, response.Code)
+			}
+			if response.Header().Get("Content-Type") != tt.expectedContentType {
+				t.Errorf("content type should be %s but got %s", tt.expectedContentType, response.Header().Get("Content-Type"))
+			}
+			if response.Body.String() != tt.expectedResponse {
+				t.Errorf("response body not matching expected is %s and received is %s", tt.expectedResponse, response.Body.String())
+			}
+		})
+	}
+}
+
+func TestHelmHandlers_HandleHelmPatchRelease(t *testing.T) {
```

Comment:
should be named `TestHelmHandlers_HandleRollbackRelease`

Reply:
Done